File: python/src/python/com/example/script/side2.py
from m8test_java.com.m8test.script.GlobalVariables import _events
from m8test_java.com.m8test.script.GlobalVariables import _logger
from m8test_java.com.m8test.script.GlobalVariables import _script

_logger.info("PYTHON_SIDE2_STARTED")

# 获取事件发布器，用于发布事件
publisher = _events.getPublisher()
# 发送本地事件，只有同一个脚本引擎启动的脚本并且本地订阅的频道和id都相同时才能收到
def publish_events(_):
    publisher.publish(lambda scopes: scopes.getEngine(), "subscription-channel",
                      lambda payload: payload.putString("data", "本地事件"))
    publisher.publish(lambda scopes: scopes.getApp(), "subscription-channel",
                      lambda payload: payload.putString("data", "全局事件"))
    _logger.info("PYTHON_SIDE2_EVENTS_PUBLISHED")

_script.getThreads().getMain().getTimer().setTimeout(publish_events, 1000)

File: php/src/com/example/script/side2.php
<?php
/** @var m8test_java\com\m8test\script\core\api\logger\Logger $logger */
/** @var m8test_java\com\m8test\script\core\api\event\Events $events */
global $logger;
global $events;
$logger->info("PHP side2 started");
// 获取事件发布器，用于发布事件
$publisher = $events->getPublisher();
// 同一个 Engine 中的脚本使用 engine scope 互通事件。
global $script;
$timer = $script->getThreads()->getMain()->getTimer();
$timer->setTimeout(function ($it) use ($publisher) {
    $publisher->publish(function ($scopes) {
        return $scopes->getEngine();
    }, "subscription-channel", function ($payload) {
        $payload->putString("message", "来自 PHP side2");
    });
    global $logger;
    $logger->info("PHP side2 published event");
}, 500);
